lib/search/embeddings/bumblebee_provider.ex

```diff
@@ -0,0 +1,128 @@
+defmodule Search.Embeddings.BumblebeeProvider do
+  alias Search.{Embeddings, Packages, Repo}
+  import Ecto.Query
+
+  @behaviour Embeddings.Provider
+
+  @impl true
+  def child_spec(opts) do
+    %{serving_name: serving_name} = parse_opts(opts)
+
+    {Nx.Serving, serving: load_model(opts), name: serving_name, batch_size: get_batch_size(opts)}
+  end
+
+  @impl true
+  def embed_one(text, opts) do
+    %{serving_name: serving_name} = parse_opts(opts)
+
+    Nx.Serving.batched_run(serving_name, text).embedding
+  end
+
+  @impl true
+  def embed(progress_callback, opts) do
+    %{table_name: table_name, serving_name: serving_name} =
+      parse_opts(opts)
+
+    fragments =
+      from f in Packages.DocFragment,
+        left_join: e in ^{table_name, Embeddings.Embedding},
+        on: e.doc_fragment_id == f.id,
+        where: is_nil(e)
+
+    fragments = Repo.all(fragments)
```

Comment:
This logic is not specific to the provider, we would need to do the same for any other.

If we are fine loading all fragments into memory, we it can be `embed_many(list, progress_callback, opts)`.

An alternative would be a stream API `embed_stream(stream, opts)`, where stream items are `{text :: String.t(), meta :: term()}` and the function returns a stream of `list({embedding, meta :: term()})`.